src/framework/xr/xr-anchors.js

```diff
@@ -0,0 +1,236 @@
+import { EventHandler } from '../../core/event-handler.js';
+import { platform } from '../../core/platform.js';
+import { XrAnchor } from './xr-anchor.js';
+
+/**
+ * Callback used by {@link XrAnchors#create}.
+ *
+ * @callback XrAnchorCreate
+ * @param {Error|null} err - The Error object if failed to create an anchor or null.
+ * @param {XrAnchor|null} anchor - The anchor that is tracked against real world geometry.
+ */
+
+/**
+ * Anchors provide an ability to specify a point in the world that need to be updated to
+ * correctly reflect the evolving understanding of the world by the underlying AR system,
+ * such that the anchor remains aligned with the same place in the physical world.
+ * Anchors tend to persist better relative to the real world, especially during a longer
+ * session with lots of movement.
+ *
+ * @property {boolean} supported True if Anchors are supported.
+ * @property {XrAnchor[]} list Array of active {@link XrAnchor}s.
+ * @augments EventHandler
+ * @category XR
+ * ```javascript
+ * app.xr.start(camera, pc.XRTYPE_AR, pc.XRSPACE_LOCALFLOOR, {
+ *     anchors: true
+ * });
+ * ```
+ */
+class XrAnchors extends EventHandler {
+    /**
+     * @type {boolean}
+     * @private
+     */
+    _supported = platform.browser && !!window.XRAnchor;
+
+    /**
+     * List of anchor creation requests.
+     *
+     * @type {Array<object>}
+     * @private
+     */
+    _creationQueue = [];
+
+    /**
+     * @type {Map<XRAnchor,XrAnchor>}
+     * @ignore
```

Comment:
Why isn't this `@private`?

Reply:
It is accessed from xr-anchor.js.

Reply:
Ah. Well, the TypeScript compiler doesn't know what an XRAnchor is, and because it's just ignored by the docs but still technically public API, it's in the types. Which is why the TS declarations are failing.

Reply:
Is there a way to provide a list of "global" classes, so TS will know about it? I've looked and couldn't find.

Reply:
I think this problem is similar to what was faced with `XrInputSource` here:

https://github.com/playcanvas/engine/blob/main/src/framework/xr/xr-input-source.js#L146

The type was just set to `*` to get around the problem.

Reply:
Maybe this could help too: https://github.com/playcanvas/engine/pull/5556

Reply:
Can verify - it helps.

Reply:
Interesting. I've merged that PR now so feel free to merge main to here.